private/buf/cmd/buf/command/generate/generate.go

```diff
@@ -399,38 +489,111 @@ func run(
 	if wasmEnabled {
 		generateOptions = append(
 			generateOptions,
-			bufgenv1.GenerateWithWASMEnabled(),
+			bufgen.GenerateWithWASMEnabled(),
 		)
 	}
-	var includedTypes []string
-	if len(flags.Types) > 0 || len(flags.TypesDeprecated) > 0 {
-		// command-line flags take precedence
-		includedTypes = append(flags.Types, flags.TypesDeprecated...)
-	} else if genConfig.TypesConfig != nil {
-		includedTypes = genConfig.TypesConfig.Include
-	}
-	if len(includedTypes) > 0 {
-		image, err = bufimageutil.ImageFilteredByTypes(image, includedTypes...)
-		if err != nil {
-			return err
-		}
-	}
 	wasmPluginExecutor, err := bufwasm.NewPluginExecutor(
-		filepath.Join(container.CacheDirPath(), bufcli.WASMCompilationCacheDir))
+		filepath.Join(
+			container.CacheDirPath(),
+			bufcli.WASMCompilationCacheDir,
+		),
+	)
 	if err != nil {
 		return err
 	}
-	return bufgenv1.NewGenerator(
+	generator := bufgen.NewGenerator(
 		logger,
 		storageosProvider,
 		runner,
 		wasmPluginExecutor,
 		clientConfig,
-	).Generate(
+	)
+	for _, image := range inputImages {
```

Comment:
This represents a change in behavior for v1 - previously, we were merging the image and then calling `Generate` once, now we are looping through resultant images. This will result in extra calls to certain plugins that cannot tolerate them

Reply:
Actually, the behavior for `v1` config file is exactly the same:
in the switch case above (when version is `v1`), only one image is added to the list, and this image is already the result of a merge, see `getInputImage`: `image, err := bufimage.MergeImages(images...)`.

In a higher level, each input (one `input` from v2 config counts as one input, and the command line argument also counts as one input) corresponds to one `image` in this `for-loop`. It's worth noting that `getInputImage` transforms one `Ref` to one `Image`.